feature-libs/product/multi-dimensional/core/services/product-multi-dimensional-images.service.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import {
+  Config,
+  Image,
+  OccConfig,
+  VariantOptionQualifier,
+  VariantQualifier,
+} from '@spartacus/core';
+
+@Injectable({ providedIn: 'root' })
+export class ProductMultiDimensionalImagesService {
+  protected config: Config = inject(Config);
+
+  /**
+   * Retrieves the images for variant option qualifiers that match the specified format.
+   */
+  getVariantOptionImages(
+    variantOptionQualifiers: VariantOptionQualifier[],
+    variantOptionValue: string
+  ): Image[] {
+    const format = VariantQualifier.SWATCH;
+    return variantOptionQualifiers
+      .filter((optionQualifier) => optionQualifier.image?.format === format)
+      .map((optionQualifier) => {
+        const altText = optionQualifier.image?.altText ?? variantOptionValue;
+        return {
+          altText,
+          format: optionQualifier.image?.format,
+          url: this.getBaseUrl() + optionQualifier.image?.url,
+        };
+      });
+  }
+
+  protected getBaseUrl(): string {
+    return (
+      (this.config as OccConfig).backend?.media?.baseUrl ??
+      (this.config as OccConfig).backend?.occ?.baseUrl ??
```

Comment:
[MINOR]
Is this type casting really needed? I believe `Config` it's already well-typed with `backend` property thanks to module-augmentation which we already have in place for `OccConfig`.

Reply:
fixed